Civi/Api4/Provider/ActionObjectProvider.php

```diff
@@ -48,6 +50,10 @@ public static function getSubscribedEvents() {
   public function onApiResolve(ResolveEvent $event) {
     $apiRequest = $event->getApiRequest();
     if ($apiRequest instanceof AbstractAction) {
+      if (!isset($this->getEntities()[$apiRequest->getEntityName()])) {
```

Comment:
This gives a slightly more informative warning if the mismatch occurs. You get `Api not implemented` rather than `DB table does not exist`.

Possibly we could throw the error right here - I wasn't sure if there are legitimate cases where something else might provide the Api.

Reply:
Nope, that's it. That function invokes `'civi.api4.entityTypes'` which is the only way to add extra entities, so this is good.

Reply:
Changed this from a log message to an exception and... it "fixes" a bunch of fails in the upgrade test, because Exceptions get swallowed e.g. when reconciling ManagedEntities

Which isn't totally satisfactory, but I guess maybe why it's not failing more currently